pkg/core/core.go

```diff
@@ -222,6 +222,8 @@ func (c *Core) Run(ctx context.Context, id uint64, opts models.RunOptions) model
 		if appErr.Err != nil {
 			utils.LogError(c.logger, appErr, "error while running the app")
 			appErrCh <- appErr
+		} else if appErr.AppErrorType == models.ErrCtxCanceled {
+			c.Proxy.SetUserAppTerminateInitiated(true)
```

Comment:
include app id in it. Use https://github.com/keploy/keploy/blob/e35909158d3728b25d58304be55ba1ff7c7e2972/pkg/core/service.go#L113 for storing where the app is closed or not. May be create a channel param for it.